### PR TITLE
libqedr: Add support for doorbell overflow recovery in user space

### PR DESCRIPTION
**Pull request by mkalderon**
Sending a patch series to rdma-next: 
Use the doorbell overflow recovery mechanism for RDMA 
commit 36907cd5cd72 ("qed: Add doorbell overflow recovery mechanism")
for rdma ( RoCE and iWARP )

this is the user-side implementation for it.

### CODE REVIEW

File: kernel-headers/rdma/qedr-abi.h [hunk 1]
```diff
@@ -68,6 +68,7 @@ struct qedr_alloc_pd_uresp {
 struct qedr_create_cq_ureq {
 	__aligned_u64 addr;
 	__aligned_u64 len;
+	__aligned_u64 db_rec_addr;
```
[the remark this answers is not in the document]
Comment (mkalderon) [reply kernel-headers/rdma/qedr-abi.h]: Sure. I thought you usually want to review the kernel + rdma-core changes at the same time. should 
I have only sent the rdma-core patches to the mailing list for review without the pull request ?

**Comment by mkalderon** [comment]
Hi Jason,
Sorry for the delay. 
Following your comment on the kernel side it seems we can have a much more elegant solution using get_user for reading the doorbell shadow instead of mapping the memory. 
This required another patch series in net which hasn't been sent yet. 
I'll cancel this pull request as the kernel changes will affect this pull request as well. 
thanks,
Michal

**Comment by mkalderon** [comment]
Hi Jason, 
Sorry for the hassle. Went back to original solution with minor changes in kernel. I'm sending a V2 for rdma-next and re-opening this pull request,

thanks,
Michal

**Comment by mkalderon** [comment]
the only failure i saw in the travis-ci details is written below and doesn't seem related to the patch. 
There is a warning on the patch description - but it doesn't look like its causing the travis-ci to fail. 
am i missing something? 

thanks,
Michal

Patch from '/home/travis/build/linux-rdma/rdma-core/buildlib/sparse-include/27/stdlib.h.diff' failed
Unable to apply any patch to 'stdlib.h', tries 1
Patch from '/home/travis/build/linux-rdma/rdma-core/buildlib/sparse-include/27/netinet-in.h.diff' failed
Patch from '/home/travis/build/linux-rdma/rdma-core/buildlib/sparse-include/25/netinet-in.h.diff' failed
Traceback (most recent call last):
  File "/home/travis/build/linux-rdma/rdma-core/buildlib/gen-sparse.py", line 159, in <module>
    replace_header(I);
  File "/home/travis/build/linux-rdma/rdma-core/buildlib/gen-sparse.py", line 103, in replace_header
    if apply_patch(find_system_header(args,fn),
  File "/home/travis/build/linux-rdma/rdma-core/buildlib/gen-sparse.py", line 48, in find_system_header
    raise ValueError("Could not find system include directory.");
ValueError: Could not find system include directory.
CMake Error at buildlib/RDMA_Sparse.cmake:28 (message):
  glibc header file patching for sparse failed.  Review include/*.rej and fix
  the rejects, then do
  /home/travis/build/linux-rdma/rdma-core/buildlib/gen-sparse.py -out
  /home/travis/build/linux-rdma/rdma-core/build-sparse/include/ --src
  /home/travis/build/linux-rdma/rdma-core/ --save
Call Stack (most recent call first):
  CMakeLists.txt:188 (RDMA_CheckSparse)
-- Configuring incomplete, errors occurred!
See also "/home/travis/build/linux-rdma/rdma-core/build-sparse/CMakeFiles/CMakeOutput.log".
See also "/home/travis/build/linux-rdma/rdma-core/build-sparse/CMakeFiles/CMakeError.log".
The command "buildlib/travis-build" exited with 1.

**Comment by jgunthorpe** [comment]
What is up with the PR? Did the kernel part ever get resent?

**Comment by mkalderon** [comment]
> What is up with the PR? Did the kernel part ever get resent?

Hi Jason, 
yes, I sent a V3 of the kernel part last week:
https://patchwork.kernel.org/patch/10992685/
https://patchwork.kernel.org/patch/10992679/
https://patchwork.kernel.org/patch/10992677/

thanks,
Michal

**Pull request closed without merging**